data/transactions/logic/assembler_test.go

```diff
@@ -228,14 +235,17 @@ func TestAssemble(t *testing.T) {
 		// Ensure that we have some basic check of all the ops, except
 		// we don't test every combination of
 		// intcblock,bytecblock,intc*,bytec*,arg* here.
-		if !strings.Contains(bigTestAssembleNonsenseProgram, spec.Name) &&
+		if !strings.Contains(v1Nonsense+v2Nonsense, spec.Name) &&
 			!strings.HasPrefix(spec.Name, "int") &&
 			!strings.HasPrefix(spec.Name, "byte") &&
-			!strings.HasPrefix(spec.Name, "arg") {
+			!strings.HasPrefix(spec.Name, "arg") &&
+			spec.Version <= 2 {
 			t.Errorf("test should contain op %v", spec.Name)
 		}
 	}
-	ops, err := AssembleStringWithVersion(bigTestAssembleNonsenseProgram, AssemblerMaxVersion)
+	// First, we test v2, not AssemblerMaxVersion. A higher version is
+	// allowed to differ (and must, in the first byte).
+	ops, err := AssembleStringWithVersion(v1Nonsense+v2Nonsense, 2)
```

Comment:
I think that this type of test would be better implemented with a TEAL fuzzer, creating random TEAL programs, assembling and disassembling them. ( the rationale here is that the nonsense programs you've picked here covers only small number of opcodes. we need to check all the opcodes being assembled/disassembled )

Reply:
Agreed that fuzzer would be a nice addition.  But this test is better than you think - it actually does require that every opcode that is available actually appears. That was @brianolson , I assume?  Anyway, point taken that adding fuzzing would be nice, but I like this test as at least a quick sanity that every opcode works in at least one simple use case.